src/test/controllers/Trip.test.js

```diff
@@ -0,0 +1,292 @@
+      sinon.stub(models.Trips, 'update').throws();
+  
+      await updateTrip(req, res);
+      expect(res.status).to.have.been.calledWith(500);
+    });
+  });
+
+  describe(' VALIDATE AUTHENTICATION ', () => {
+    it('Should return 401 if user is not a validated user with a role on the application', (done) => {
+      const notaValidUser = {
+        email: 'dele@gmail.com',
+        password: 'dele1234'
+      };
+      chai
+        .request(server)
+        .post(signinRoute)
+        .send(notaValidUser)
+        .end((logError, logResponse) => {
+          const token = `Bearer ${logResponse.body.data.token}`;
+          const trip = {
+            origin: 'lagos',
+            destination: 'nairobi',
+            departureDate: '2019-09-09 01:33:28.862+00',
+            travelReasons: 'To get away.',
+            typeOfTrip: 'Return',
+            roomId: 'b098fb80-72c7-4a9a-a35a-634692384d89',
+            accommodationId: '2b770fbc-76e6-4b5a-afab-882759fd1f06'
+          };
+          chai.request(server)
+            .post('/api/v1/trip/47db7b6c-394f-452b-a09d-19bff77f84d9')
+            .set('Authorization', token)
+            .send(trip)
+            .end((error, response) => {
+              expect(response).to.have.status(401);
+              expect(response.body).to.have.property('status');
+              expect(response.body).to.have.property('message');
+              done();
+            });
+        });
+    });
+    it('Should return 401 if user is not manager or super admin', (done) => {
+      const notaValidRole = {
+        email: 'kelechi@gmail.com',
+        password: 'kelechi'
+      };
+      chai
+        .request(server)
+        .post(signinRoute)
+        .send(notaValidRole)
+        .end((logError, logResponse) => {
+          const token = `Bearer ${logResponse.body.data.token}`;
+          chai.request(server)
+            .get('/api/v1/request/manager')
+            .set('Authorization', token)
+            .end((error, response) => {
+              expect(response).to.have.status(401);
+              expect(response.body).to.have.property('status');
+              expect(response.body).to.have.property('message');
+              done();
+            });
+        });
+    });
+    it('should fake an internal server error in verifying the users', async () => {
+      const req = {
+        user: {
+          user: { userId: '36da1ce3-8e1f-4b0f-8e15-1189d8231ef2' }
+        }
+      };
+      const res = {
+        status: () => {},
+        json: () => {},
+      };
+      sinon.stub(res, 'status').returnsThis();
+      sinon.stub(models.Trips, 'findOne').throws();
+      sinon.stub(models.Requests, 'findOne').throws();
+  
+      await Authenticate.verifyUser(req, res);
+      expect(res.status).to.have.been.calledWith(500);
+      await Authenticate.verifyRequest(req, res);
+      expect(res.status).to.have.been.calledWith(500);
+    });
+    // it('dept', async () => {
```

Comment:
@funmi5 you should also remove this commented code out

Reply:
OK.